storage/innobase/row/row0merge.cc

```diff
@@ -2029,7 +2029,6 @@ row_merge_read_clustered_index(
 					row = NULL;
 					mtr_commit(&mtr);
 					mem_heap_free(row_heap);
-					ut_free(nonnull);
 					goto write_buffers;
```

Comment:
I do not think that this is correct. The alleged double free of `nonnull` is executed later in the code, but only if `row != NULL`. Here is the code from 10.0, which is the oldest applicable version:
```c++
		if (row == NULL) {
			goto all_done;
		}

		if (err != DB_SUCCESS) {
			goto func_exit;
		}
	}

func_exit:
	mtr_commit(&mtr);

	mem_heap_free(row_heap);

	if (nonnull) {
		mem_free(nonnull);
	}

all_done:
```